src/lavinmq/mqtt/retain_store.cr

```diff
@@ -0,0 +1,144 @@
+require "./topic_tree"
+require "digest/sha256"
+
+module LavinMQ
+  module MQTT
+    class RetainStore
+      Log = LavinMQ::Log.for("retainstore")
+
+      MESSAGE_FILE_SUFFIX = ".msg"
+      INDEX_FILE_NAME     = "index"
+
+      alias IndexTree = TopicTree(String)
+
+      def initialize(@dir : String, @index = IndexTree.new)
+        Dir.mkdir_p @dir
+        @index_file = File.new(File.join(@dir, INDEX_FILE_NAME), "a+")
```

Comment:
Is this file replicated? Should be?

Reply:
RetainStore is copied from Myra. Probably a lot that can be improved, like using MFile.